climada/hazard/wildfire.py

```diff
@@ -0,0 +1,1165 @@
+"""
+This file is part of CLIMADA.
+
+Copyright (C) 2017 ETH Zurich, CLIMADA contributors listed in AUTHORS.
+
+CLIMADA is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free
+Software Foundation, version 3.
+
+CLIMADA is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along
+with CLIMADA. If not, see <https://www.gnu.org/licenses/>.
+
+---
+
+Define WildFire class.
+"""
+
+__all__ = ['WildFire']
+
+import logging
+import warnings
+import itertools
+from datetime import date
+from functools import partial
+import numpy as np
+import matplotlib.pyplot as plt
+import pandas as pd
+from scipy import sparse
+from sklearn.neighbors import BallTree
+from sklearn.cluster import DBSCAN
+import numba
+import geopandas as gpd
+from shapely.geometry import Point
+from shapely.ops import transform
+import pyproj
+
+from climada.hazard.centroids.centr import Centroids
+from climada.hazard.base import Hazard
+from climada.hazard.tag import Tag as TagHazard
+from climada.util.constants import ONE_LAT_KM, DEF_CRS
+from climada.util.dates_times import str_to_date
+
+from climada.util.alpha_shape import alpha_shape, plot_polygon
```

Comment:
As a convention in CLIMADA, it is recommended to  import util modules with names 'u_' in order to identify them clearly in the code. Here for example
```suggestion
import climada.util.dates_times as u_dt

import climada.util.alpha_shape as u_alpha
```
and then adapt the code for `str_to_date` -> `u_dt.str_to_date`, `alpha_shape` -> `u_alpha.alpha_shape`, `plot_polygon` -> `u_alpha.plot_polygon`.

Reply:
done - commit 21c01dc3